tenseal/binding.cpp

```diff
@@ -141,6 +144,10 @@ PYBIND11_MODULE(_tenseal_cpp, m) {
         .def("__sub__", py::overload_cast<double>(&CKKSVector::sub_plain))
         .def("__sub__",
              py::overload_cast<vector<double>>(&CKKSVector::sub_plain))
+        // subtraction operation is anticommutative
+        .def("__rsub__", py::overload_cast<double>(&CKKSVector::rsub_plain))
```

Comment:
I think we can define them here instead of defining an rsub method in CKKSVector

```suggestion
        .def("__rsub__", [](double left_operand) {
                 CKKSVector new_vector = this->negate();
                 new_vector.add_plain_inplace(left_operand);
                 return new_vector;
             })
```

Reply:
I thought about that before, but since we need to access `this` pointer we can do that in a lambda function.

Reply:
True! This should work:

```suggestion
        .def("__rsub__", [](CKKSVector vec, double left_operand) {
                 // vec should be a copy so it might be safe to do inplace
                 vec->negate_inplace();
                 vec.add_plain_inplace(left_operand);
                 return vec;
             })
```